### PR TITLE
[IMP] l10n_ar: We have adapted the demo data module to enhance data privacy.

### PR DESCRIPTION
**Pull request by mem-adhoc**
Task: 41734
Odoo-task: 1239

Description of the issue/feature this PR addresses:
This task was created to modify the demo data for Argentina, to avoid using real partner information except for their VAT number.

Current behavior before PR:

Desired behavior after PR is merged:
Edit the demo data with fake information except for the VAT number.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by vbe-odoo** [comment]
@mem-adhoc thanks for the PR.
Feedback:
- We need to remove the logos of ex Cerro Castor (now Montana Sur) and Consejo de Rosario (now Servicios Globales SRL)
image

All the rest is OK.

Thanks!

**Comment by mem-adhoc** [comment]
Hi @vbe-odoo!
Hello 
I've already made the changes you asked for!
image

**Comment by vbe-odoo** [comment]
@mem-adhoc thanks! It looks good now!
@alialfie can you help us with the review of this PR? Or maybe ping someone who can?

Thanks!
cc' @ren-odoo @masi-odoo @fvz-odoo @jco-odoo

**Comment by vbe-odoo** [comment]
Hello @mem-adhoc 
We viewed one more change we forgot, the contact is Expresso, can you help me removing the [logo](https://github.com/odoo/odoo/blob/2f0590778fcf7856babc1482cde8e3038df49a23/addons/l10n_ar/demo/expresso_logo.png) and changing the [email, website, and name](https://github.com/odoo/odoo/blob/2f0590778fcf7856babc1482cde8e3038df49a23/addons/l10n_ar/demo/res_partner_demo.xml#L87), new values:
- Name: Barcelona Foods
- Email: info@barcelonafoods.com
- Website: http://www.barcelonafoods.com/
- Logo: remove it

Thanks!
@jjscarafia @zaoral

**Review by alialfie — CHANGES_REQUESTED** [review]
Hi! Left a few comments below :)

Also, about the commit message, please add a short explanation as to why we're doing this change. And I couldn't find a task with id 41734. Is it the correct task?
Also, a small nitpick, but the correct format according to the guildlines would be:
```
task-{id}
```
For the `Odoo-task` part, could be nice to say that this is the latam-task id :)
Thanks!

**Comment by mem-adhoc** [comment]
Hello @vbe-odoo !
I've already made the changes you requested! I also changed the tests to make them consistent with the demo data!

**Comment by vbe-odoo** [comment]
@mem-adhoc there's an error in the automated tests `ci/minimal` due to the title of the commit message:
**Error:**

> Invalid title for commit: 'This task was created to modify the demo data for Argentina, to avoid using real partner information except for their VAT number.'

Can you please change the commit title to the same as in the PR title, and move the current commit title to the body/description of the commit message?

**Current Title:**
> [ADD] l10n_ar: We have adapted the demo data module to enhance data privacy

Thanks!

**Comment by vbe-odoo** [comment]
@alialfie all changes are done, can you help me with the final review and maybe merge of this?
Thanks!

File: addons/l10n_ar/tests/common.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def setUpClass(cls, chart_template_ref='l10n_ar.l10nar_ri_chart_template'):
             'vat': "27320732811",
             'l10n_ar_afip_responsibility_type_id': cls.env.ref("l10n_ar.res_RM").id,
         })
-        cls.res_partner_cerrocastor = cls.env['res.partner'].create({
+        cls.res_partner_montaña_sur = cls.env['res.partner'].create({
             "name": "Cerro Castor (Tierra del Fuego)",
```
Comment (alialfie) [inline addons/l10n_ar/tests/common.py:115]: Might as well update the partner's data like you did for the XML :)

File: addons/l10n_ar/tests/common.py [hunk 2]
```diff
@@ -125,7 +125,7 @@ def setUpClass(cls, chart_template_ref='l10n_ar.l10nar_ri_chart_template'):
             'vat': "27333333339",
             'l10n_ar_afip_responsibility_type_id': cls.env.ref("l10n_ar.res_IVA_LIB").id,
         })
-        cls.res_partner_cmr = cls.env['res.partner'].create({
+        cls.res_partner_servicios_globales = cls.env['res.partner'].create({
             "name": "Concejo Municipal de Rosario (IVA Sujeto Exento)",
```
Comment (alialfie) [inline addons/l10n_ar/tests/common.py:129]: idem

File: addons/l10n_ar/tests/common.py [hunk 3]
```diff
@@ -140,7 +140,7 @@ def setUpClass(cls, chart_template_ref='l10n_ar.l10nar_ri_chart_template'):
             'vat': "30684679372",
             'l10n_ar_afip_responsibility_type_id': cls.env.ref("l10n_ar.res_IVAE").id,
         })
-        cls.res_partner_expresso = cls.env['res.partner'].create({
+        cls.res_partner_barcelona_food = cls.env['res.partner'].create({
             "name": "Expresso",
```
Comment (alialfie) [inline addons/l10n_ar/tests/common.py:144]: idem

File: addons/l10n_ar/tests/common.py [hunk 4]
```diff
@@ -396,7 +396,7 @@ def _create_test_invoices_like_demo(self, use_current_date=True):
             },
             'test_invoice_6': {
                 'ref': 'test_invoice_6: Invoice to cerro castor, fiscal position changes taxes to exempt',
-                "partner_id": self.res_partner_cerrocastor,
+                "partner_id": self.res_partner_montaña_sur,
```
Comment (alialfie) [inline addons/l10n_ar/tests/common.py:399]: idem

File: addons/l10n_ar/tests/test_manual.py [hunk 5]
```diff
@@ -39,11 +39,11 @@ def test_02_fiscal_position(self):
         self.assertFalse(invoice.fiscal_position_id, 'Fiscal position should be set to empty')
 
         # Cerro Castor > IVA Liberado – Ley Nº 19.640 > Compras / Ventas Zona Franca > IVA Exento
-        invoice = self._create_invoice({'partner': self.res_partner_cerrocastor})
+        invoice = self._create_invoice({'partner': self.res_partner_montaña_sur})
```
Comment (alialfie) [inline addons/l10n_ar/tests/test_manual.py:42]: idem

**Review by alialfie — CHANGES_REQUESTED** [review]
Hi! Thanks for the updates :)

For the commit message, please follow the [guidelines](https://www.odoo.com/documentation/17.0/contributing/development/git_guidelines.html).

For the rest, I left some comments down below. Thanks!

File: addons/l10n_ar/tests/common.py [hunk 6]
```diff
@@ -111,7 +111,7 @@ def setUpClass(cls, chart_template_ref='l10n_ar.l10nar_ri_chart_template'):
             'vat': "27320732811",
             'l10n_ar_afip_responsibility_type_id': cls.env.ref("l10n_ar.res_RM").id,
         })
-        cls.res_partner_cerrocastor = cls.env['res.partner'].create({
+        cls.res_partner_montaña_sur = cls.env['res.partner'].create({
```
[the remark this answers is not in the document]
Comment (jco-odoo) [reply addons/l10n_ar/tests/common.py:114]: @alialfie I don't think it is a problem in itself (that someone told me that xml_ids can be utf-8)

[the remark this answers is not in the document]
Comment (alialfie) [reply addons/l10n_ar/tests/common.py:114]: Wouldn't it still be better to not use special characters? Someone trying to access that variable without a specific keyboard will have to always copy and paste

**Comment by vbe-odoo** [comment]
Hello @mem-adhoc can you help me with the requested changes?
Thanks!

File: addons/l10n_ar/tests/common.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (vbe-odoo) [reply addons/l10n_ar/tests/common.py:114]: @mem-adhoc we can avoid special characters in letters to make it easier, let's pick names without (´) or (Ñ) on it. Thanks!

**Comment by mem-adhoc** [comment]
Hi @vbe-odoo!
I made the changes you asked me for!

File: addons/l10n_ar/demo/account_customer_invoice_demo.xml [hunk 7]
```diff
@@ -93,7 +93,7 @@
 
     <!-- Export invoice to expresso, fiscal position changes tax to exempt (type 4 because it have services) -->
```
[the remark this answers is not in the document]
Comment (alialfie) [reply addons/l10n_ar/demo/account_customer_invoice_demo.xml:94]: Don't forget about this :)

**Review by alialfie — COMMENTED** [review]
The changes look good, thanks!
What about the commit message? Please follow the guidelines I linked in my previous review :)

**Comment by mem-adhoc** [comment]
Hi @alialfie!
I made the changes you asked me for!
cc @vbe-odoo

**Review by alialfie — COMMENTED** [review]
Thanks @mem-adhoc , but the commit title is still not quite right. It should be more like `[IMP] l10n_ar: modify demo data` for the title. The tag `ADD` is for adding new modules :)

**Comment by mem-adhoc** [comment]
Hi @alialfie!
I made the changes you asked me for!

**Review by alialfie — CHANGES_REQUESTED** [review]
@mem-adhoc Almost there, but the commit title itself should be changed, not just the PR title :) Also, please change the commit title too, not just the tag. And there is one small style error, please check that as well. Thanks!

**Comment by mem-adhoc** [comment]
Hi @alialfie!
I made the changes you asked me for!

**Comment by alialfie** [comment]
Hey @mem-adhoc , please do the following changes:
1- for the commit message: change the title to be "[IMP] l10n_ar: modify demo data".
2- for the style error: in `test_manual.py` line 41, change the hyphen to a dash. 
image
image

**Comment by mem-adhoc** [comment]
Hi @alialfie!
I made the changes you asked me for!

**Review by alialfie — APPROVED** [review]
Thanks everyone!
LGTM @FlorianGilbert

**Comment by alialfie** [comment]
Hello @mem-adhoc 
After some discussions with @vbe-odoo and @FlorianGilbert , while a PR like this should normally be merged in master, we'll go ahead and merge this in stable but we need to keep the XML IDs as they were.

Could you please change the PR again so that the XML IDs remain unchanged? That's the only thing needed so we can merge this. Thanks!

**Comment by mem-adhoc** [comment]
Hi @alialfie!
I have already changed the XML IDs. I understand that I should leave the tests as they are.

**Review by alialfie — APPROVED** [review]
Thanks @mem-adhoc
All good now @FlorianGilbert

**Comment by mem-adhoc** [comment]
@alialfie Hi!
Could you see this ??

**Comment by jco-odoo** [comment]
@robodoo r+

**Comment by jco-odoo** [comment]
@mem-adhoc Could you check all of the tests please? (see link above)

**Comment by vbe-odoo** [comment]
@jco-odoo @alialfie this is ready for your review, all checks passed.
Thanks
cc' @mem-adhoc

**Review by zaoral — COMMENTED** [review]
+1

**Comment by pablohmontenegro** [comment]
Hello @vbe-odoo , why the FW to 18 does not exist? we also need it in 18, thanks!

**Comment by jco-odoo** [comment]
@pablohmontenegro Because it is still stuck in saas-17.2

**Pull request closed without merging**